packages/core/vue/src/useWallet.ts

```diff
@@ -0,0 +1,225 @@
+
+    // Handle the adapter events.
+    const onReady = () => ready.value = true;
+    const onConnect = () => {
+        if (! adapter.value) return;
+        ready.value = adapter.value.ready;
+        publicKey.value = adapter.value.publicKey;
+        connected.value = adapter.value.connected;
+    };
+    watchEffect(onInvalidate => {
+        if (! adapter.value) return;
+        adapter.value.on('ready', onReady);
+        adapter.value.on('connect', onConnect);
+        adapter.value.on('error', onError);
+        onInvalidate(() => {
+            if (! adapter.value) return;
+            adapter.value.off('ready', onReady);
+            adapter.value.off('connect', onConnect);
+            adapter.value.off('error', onError);
+        });
+    });
+
+    // Helper method to return an error whilst using the onError callback.
+    const newError = (error: WalletError): WalletError => {
+        onError(error);
+        return error;
+    };
+
+    // Connect the adapter to the wallet.
+    const connect = async (): Promise<void> => {
+        if (connected.value || connecting.value || disconnecting.value) return;
+        if (! wallet.value || ! adapter.value) throw newError(new WalletNotSelectedError());
+
+        if (! ready.value) {
+            walletName.value = null;
+            window.open(wallet.value.url, '_blank');
+            throw newError(new WalletNotReadyError());
+        }
+
+        try {
+            connecting.value = true;
+            await adapter.value.connect();
+        } catch (error: any) {
+            walletName.value = null;
+            throw error;
+        } finally {
+            connecting.value = false;
+        }
+    };
+
+    // Disconnect the adapter from the wallet.
+    const disconnect = async (): Promise<void> => {
+        if (disconnecting.value) return;
+        if (! adapter.value) {
+            walletName.value = null;
+            return;
+        }
+
+        try {
+            disconnecting.value = true;
+            await adapter.value.disconnect();
+        } finally {
+            walletName.value = null;
+            disconnecting.value = false;
+        }
+    };
+
+    // Send a transaction using the provided connection.
+    const sendTransaction = async (transaction: Transaction, connection: Connection, options?: SendTransactionOptions) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        return await adapter.value.sendTransaction(transaction, connection, options);
+    };
+
+    // Sign a transaction if the wallet supports it.
+    const signTransaction = async (transaction: Transaction) => {
+        if (! adapter.value) throw newError(new WalletNotSelectedError());
+        if (! connected.value) throw newError(new WalletNotConnectedError());
+        if (!('signTransaction' in adapter.value)) throw newError(new OperationNotSupportedByWalletError());
```

Comment:
Throwing a runtime error here is something the React API design intentionally avoids. The optional methods will be undefined if the wallet doesn't support them. This produces type safety for the developer that calls them and avoids an error the user sees.